### PR TITLE
Add async dist tests

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

**Review by gongweibao — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/dist_mnist.py [hunk 1]
```diff
@@ -66,8 +68,7 @@ def cnn_model(data):
         size=SIZE,
         act="softmax",
         param_attr=fluid.param_attr.ParamAttr(
-            initializer=fluid.initializer.NormalInitializer(
-                loc=0.0, scale=scale, seed=1)))
+            initializer=fluid.initializer.Constant(value=0.1)))
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/dist_mnist.py:71]: why change this?

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 2]
```diff
@@ -1346,7 +1328,7 @@ def _get_param_block(opt_op):
 
         for key in opt_op.input_names:
             new_shape = None
-            if key in ["Param", "Grad", "LearningRate", "Moment", "MeanSquare"]:
+            if key in ["Param", "Grad", "LearningRate"]:
```
Comment (panyx0718) [inline python/paddle/fluid/transpiler/distribute_transpiler.py:1331]: glad to see these are gone. Can you briefly explain what it was added and why we don't need it anymore?

**Review by panyx0718 — APPROVED** [review]
Nice! We need better coverage.

File: python/paddle/fluid/tests/unittests/dist_mnist.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/fluid/tests/unittests/dist_mnist.py:71]: pserver side initializer may create different values on CPU/GPU

File: python/paddle/fluid/transpiler/distribute_transpiler.py (hunk 2 above)
Comment (typhoonzero) [reply python/paddle/fluid/transpiler/distribute_transpiler.py:1331]: Well, it was some careless change in a very early time: https://github.com/PaddlePaddle/Paddle/pull/12798/files#diff-3a23ecb6a98b666c84525f3f6fcd2bb6L1212

the if condition lost the rmsprop op.

**Pull request merged**